components/brave_wallet_ui/components/extension/encryption-key-panel/index.tsx

```diff
@@ -42,18 +43,20 @@ function EncryptionKeyPanel (props: Props) {
     accounts,
     selectedNetwork,
     encryptionKeyPayload,
+    decryptPayload,
     onProvideOrAllow,
     onCancel
   } = props
   const [isDecrypted, setIsDecrypted] = React.useState<boolean>(false)
+  const payloadAddress = panelType === 'request' ? encryptionKeyPayload.address : decryptPayload.address
 
   const foundAccountName = React.useMemo(() => {
-    return accounts.find((account) => account.address.toLowerCase() === encryptionKeyPayload.address.toLowerCase())?.name
-  }, [encryptionKeyPayload])
+    return accounts.find((account) => account.address.toLowerCase() === payloadAddress.toLowerCase())?.name
+  }, [encryptionKeyPayload, decryptPayload])
 
   const orb = React.useMemo(() => {
-    return create({ seed: encryptionKeyPayload.address.toLowerCase(), size: 8, scale: 16 }).toDataURL()
-  }, [encryptionKeyPayload])
+    return create({ seed: payloadAddress.toLowerCase(), size: 8, scale: 16 }).toDataURL()
+  }, [encryptionKeyPayload, decryptPayload])
```

Comment:
Same here!

Reply:
should have realized, thx for catching
037fe8165b0359d535cff2e68250f222c1b1cde6